FA22.P02.Web/Program.cs

```diff
@@ -16,33 +18,120 @@
 
 app.UseHttpsRedirection();
 
-var summaries = new[]
+
+List<Product> products1 = new List<Product>() {
+                new Product(){ Id = 1, Name="Ketchup", Description="Condiment", Price= 4.99},
+                new Product(){ Id = 2, Name="Mustard", Description="Condiment", Price= 2.99},
+                new Product(){ Id = 3, Name="Mayo", Description="Condiment", Price= 3.99},
+                
+            };
+
+
+
+
+
+
+
+app.MapGet("/product/get-product", (int id) =>
 {
-    "Freezing", "Bracing", "Chilly", "Cool", "Mild", "Warm", "Balmy", "Hot", "Sweltering", "Scorching"
-};
+    
+    var ID = new List<Product>();
+
+    foreach (var product in products1)
+    {
+
+        if (id == product.Id)
+        {
+            ID.Add(product);
+        }
+    }
+    return ID;
+
 
-app.MapGet("/weatherforecast", () =>
+})
+    .WithName("GetProduct");
+
+
+
+app.MapGet("/products/{get-all}", () =>
 {
-    throw new Exception("remove me");
-    var forecast = Enumerable.Range(1, 5).Select(index =>
-        new WeatherForecast
-        (
-            DateTime.Now.AddDays(index),
-            Random.Shared.Next(-20, 55),
-            summaries[Random.Shared.Next(summaries.Length)]
-        ))
-        .ToArray();
-    return forecast;
+    
+        var Id = products1;
+        return Id;
+    
 })
-.WithName("GetWeatherForecast");
+    .WithName("GetProducts");
+
+app.MapPost("/products/{create-product}", (int id, string name, string description, double price) =>
+{
+
+    
+
+    
+
+    var product =  new Product(){Id = id, Name= name, Description= description, Price= price };
+    products1.Add(product);
+     var Id = products1;
+    return Id;
+
+
 
-app.Run();
 
-internal record WeatherForecast(DateTime Date, int TemperatureC, string? Summary)
+})
+    .WithName("CreateProduct");
+
+app.MapPut("/products/{update-product}", (int id, string name, string description, double price) =>
 {
-    public int TemperatureF => 32 + (int)(TemperatureC / 0.5556);
-}
+
+    var product = new Product() { Id = id, Name = name, Description = description, Price = price };
+    foreach(var products in products1)
+    {
+        if(id == products.Id)
+        {
+            
+            products.Name = name;
+            products.Description = description;
+            products.Price = price;
+        }
+    }
+    
+    var Id = products1;
+    return Id;
+})
+    .WithName("UpdateProduct");
+
+app.MapDelete("/products/{delete-product}", (int Id) =>
+{
+   
+   
+    var id = products1.FindIndex(products1 => products1.Id == Id);
+    products1.RemoveAt(id);
+
+    
+
+    var iD = products1;
+    return iD;
+
+
+})
+    .WithName("DeleteProduct");
+
+
+
+
+
+app.Run();
+
+
 
 //see: https://docs.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-6.0
 // Hi 383 - this is added so we can test our web project automatically. More on that later
-public partial class Program { }
```

Comment:
@jordbutler Removing Program broke the tests

Reply:
@jordbutler You removed a class called "Program" then added a new class called "Project" - that will not resolve the issue